trino-aws-proxy-spi/src/main/java/io/trino/aws/proxy/spi/security/SecurityResponse.java

```diff
@@ -17,12 +17,32 @@
 
 import static java.util.Objects.requireNonNull;
 
-public record SecurityResponse(boolean canProceed, Optional<String> error)
+public sealed interface SecurityResponse
```

Comment:
Wondering if we can add some an annotation like `@SuppressWarnings("checkstyle:interfacesstype")` to bypass the checkstyle here rather than disabling it

Reply:
I will try this and let you know if it works

Reply:
That annotation does work but I prefer disabling it. I've submitted a PR to airlift to remove this module. In modern Java, I think the purpose of this check is wrong. An interface to hold only constants is a good thing now.

Reply:
The project build for me without the checkstyle exclusion in the pom or the annotation. I actually can't get the `InterfaceIsType` rule to trigger at all by adding new empty interfaces. Should we undo the pom change?

Reply:
Remove it and then run: `./mvnw validate` to see the error.

Reply:
I tried, but its still building successfully 🤷‍♂️ In any case I trust that it should be excluded (and i agree it doesn't make much sense in modern java)

> ```
> ❯ ./mvnw --version
> Apache Maven 3.9.6 (bc0240f3c744dd6b6ec2920b3cd08dcc295161ae)
> Maven home: /Users/mosiac/.m2/wrapper/dists/apache-maven-3.9.6-bin/3311e1d4/apache-maven-3.9.6
> Java version: 22, vendor: Eclipse Adoptium, runtime: /Users/mosiac/.sdkman/candidates/java/22-tem
> Default locale: en_GB, platform encoding: UTF-8
> OS name: "mac os x", version: "14.5", arch: "aarch64", family: "mac"
> ```

Reply:
Wow - that's odd. I'll try to see what's happening.

Reply:
¯\\_(ツ)_/¯